cumulusci/core/flowrunner.py

```diff
@@ -581,3 +586,96 @@ def _find_result_by_path(self, path):
             if result.path[-len(path) :] == path:
                 return result
         raise NameError("Path not found: {}".format(path))
+
+
+class PreflightFlowCoordinator(FlowCoordinator):
+    """Coordinates running preflight checks instead of the actual flow steps.
+    """
+
+    def run(self, org_config):
+        self.org_config = org_config
+        self.callbacks.pre_flow(self)
+
+        self._init_org()
+        self._rule(fill="-")
+        self.logger.info("Organization:")
+        self.logger.info("  {}: {}".format("Username", org_config.username))
+        self.logger.info("  {}: {}".format("  Org Id", org_config.org_id))
+        self._rule(fill="-", new_line=True)
+
+        self.logger.info("Running preflight checks...")
+        self._rule(new_line=True)
+
+        self.jinja2_context = {
+            "tasks": TaskCache(self, self.project_config, org_config),
+            "project_config": self.project_config,
+            "org_config": self.org_config,
+        }
+
+        self.preflight_results = defaultdict(list)
+        try:
+            # flow-level checks
+            for check in self.flow_config.checks or []:
+                result = self.evaluate_check(check)
+                if result:
+                    self.preflight_results[None].append(result)
+
+            # Step-level checks
+            for step in self.steps:
+                for check in step.task_config.get("checks", []):
+                    result = self.evaluate_check(check)
+                    if result:
+                        self.preflight_results[step.path].append(result)
+        finally:
+            self.callbacks.post_flow(self)
+
+    def evaluate_check(self, check):
+        self.logger.info("Evaluating check: {}".format(check["when"]))
+        expr = jinja2_env.compile_expression(check["when"])
+        value = bool(expr(**self.jinja2_context))
+        self.logger.info("Check result: {}".format(value))
+        if value:
+            return {"status": check["action"], "message": check.get("message")}
```

Comment:
I often like the explicit `return None` rather than falling off the edge, but my Python style is not necessarily mainstream. Does it matter / is there an imperative or style rec to handle it this way?

Reply:
I don't think I've seen a particular recommendation. Personally I almost never use an explicit `return None` but my reason is mostly "it's more concise and I know what it means" which isn't a great reason.

I think part of the answer is probably to start using type checks so that we'll be forced to review anywhere None is returned inadvertently.